src/rules/objectLiteralShorthandRule.ts

```diff
@@ -15,33 +15,70 @@
  * limitations under the License.
  */
 
-import { getChildOfKind, hasModifier, isFunctionExpression, isIdentifier, isPropertyAssignment } from "tsutils";
+import {
+    getChildOfKind,
+    getModifier,
+    hasModifier,
+    isFunctionExpression,
+    isIdentifier,
+    isMethodDeclaration,
+    isPropertyAssignment,
+    isShorthandPropertyAssignment,
+} from "tsutils";
 import * as ts from "typescript";
-import * as Lint from "../index";
+import * as Lint from "..";
+
+const OPTION_NEVER = "never";
 
 export class Rule extends Lint.Rules.AbstractRule {
     /* tslint:disable:object-literal-sort-keys */
     public static metadata: Lint.IRuleMetadata = {
         ruleName: "object-literal-shorthand",
-        description: "Enforces use of ES6 object literal shorthand when possible.",
+        description: "Enforces/disallows use of ES6 object literal shorthand.",
         hasFix: true,
-        optionsDescription: "Not configurable.",
-        options: null,
-        optionExamples: [true],
+        optionsDescription: Lint.Utils.dedent`
+        If the \'never\' option is provided, any shorthand object literal syntax will cause a failure.`,
+        options: {
+            type: "string",
+            enum: [OPTION_NEVER],
+        },
+        optionExamples: [true, [true, OPTION_NEVER]],
         type: "style",
         typescriptOnly: false,
     };
     /* tslint:enable:object-literal-sort-keys */
 
     public static LONGHAND_PROPERTY = "Expected property shorthand in object literal ";
     public static LONGHAND_METHOD = "Expected method shorthand in object literal ";
+    public static SHORTHAND_ASSIGNMENT = "Shorthand property assignments have been disallowed.";
 
     public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
-        return this.applyWithFunction(sourceFile, walk);
+        return this.applyWithFunction(
+            sourceFile,
+            this.ruleArguments.indexOf(OPTION_NEVER) === -1
+                ? enforceShorthandWalker
+                : disallowShorthandWalker,
+        );
     }
 }
 
-function walk(ctx: Lint.WalkContext<void>) {
+function disallowShorthandWalker(ctx: Lint.WalkContext<void>) {
+    return ts.forEachChild(ctx.sourceFile, function cb(node): void {
+        if (isShorthandAssignment(node)) {
+            ctx.addFailureAtNode(
+                node,
```

Comment:
looking at the tests, I think it's better to just show the error at `node.name`

Reply:
Went ahead and updated the error messages for shorthand-only too